### PR TITLE
Finance: custom labels

### PR DESCRIPTION
**Pull request by AquiGorka**
Update March 20th:

These changes implement custom labels for the Finance app and depend on [@aragon.js#253](https://github.com/aragon/aragon.js/pull/253), [@aragon/ui#327](https://github.com/aragon/aragon-ui/pull/327) and [@aragon#630](https://github.com/aragon/aragon/pull/630)

## Todo

- [x] Bump minor version of `@aragon/ui` once released with updated IdentityBadge
- [x] Bump version of `@aragon/api` once released with identities functionality

### CODE REVIEW

File: apps/finance/app/src/components/CustomLabelIdentityBadge/CustomLabelIdentityBadge.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+  }
+  React.useEffect(() => {
+    handleResolve(address)
+    const subscription = updates$.subscribe(addr => {
+      if (addr === address) {
+        handleResolve(address)
+      }
+    })
+    return () => subscription.unsubscribe()
+  }, [])
```
Comment (2color) [inline apps/finance/app/src/components/CustomLabelIdentityBadge/CustomLabelIdentityBadge.js]: Same question as in https://github.com/aragon/aragon-apps/pull/722/files#r267302393

Comment (AquiGorka) [reply apps/finance/app/src/components/CustomLabelIdentityBadge/CustomLabelIdentityBadge.js]: Addresed there 😄

File: apps/finance/app/src/components/LocalIdentityBadge/LocalIdentityBadge.js [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+  }
+  React.useEffect(() => {
+    handleResolve()
+    const subscription = updates$.subscribe(updatedAddress => {
+      if (updatedAddress.toLowerCase() === address.toLowerCase()) {
+        handleResolve()
+      }
+    })
+    return () => subscription.unsubscribe()
+  }, [])
```
Comment (bpierre) [inline apps/finance/app/src/components/LocalIdentityBadge/LocalIdentityBadge.js]: Shouldn’t we re-run this if `address` changes?

File: apps/finance/app/src/components/LocalIdentityModal/LocalIdentityModalManager.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { Subject } from 'rxjs'
+
+const updates$ = new Subject()
```
Comment (bpierre) [inline apps/finance/app/src/components/LocalIdentityModal/LocalIdentityModalManager.js]: Same thing than [in the `aragon/aragon` PR](https://github.com/aragon/aragon/pull/656#discussion_r268710971) about RxJS, but I think it is especially relevant for apps as they also serve as examples for app developers.

File: apps/finance/app/src/components/LocalIdentityBadge/LocalIdentityBadge.js (hunk 2 above)
Comment (AquiGorka) [reply apps/finance/app/src/components/LocalIdentityBadge/LocalIdentityBadge.js]: For a single IdentityBadge address won't change.

Comment (bpierre) [reply apps/finance/app/src/components/LocalIdentityBadge/LocalIdentityBadge.js]: I wonder if we shouldn’t still make sure our effects always define their dependencies even if the context doesn’t requires it? The same way we always clean up our events or timers, even for single life components, just to make it explicit and easier to move code. I know the React team plans to / has already implemented this fix in their [ESLint plugin for hooks](https://github.com/facebook/react/issues/14920) (by filling this array automatically), so maybe we could anticipate.

File: apps/finance/app/src/components/LocalIdentityModal/LocalIdentityModalManager.js (hunk 3 above)
Comment (AquiGorka) [reply apps/finance/app/src/components/LocalIdentityModal/LocalIdentityModalManager.js]: I'll open an issue to keep track of this (and for all the apps too).

File: apps/finance/app/package.json [hunk 4]
```diff
@@ -4,22 +4,22 @@
...
+    "react": "^16.8.4",
     "react-display-name": "^0.2.3",
-    "react-dom": "^16.2.0",
+    "react-dom": "^16.8.4",
     "react-spring": "^7.2.8",
-    "rxjs": "^6.2.1",
+    "rxjs": "^6.4.0",
     "seed-random": "^2.2.0",
-    "styled-components": "4.1.3",
+    "styled-components": "^4.1.3",
```
Comment (sohkai) [inline apps/finance/app/package.json]: I noticed the other apps also have this pinned, and I think when I asked @bpierre he said those could also be unpinned? Should we make this unpinned for all the apps?

File: apps/finance/app/src/components/TransferRow.js [hunk 5]
```diff
@@ -121,6 +129,14 @@ class TransferRow extends React.PureComponent {
   }
 }
 
+const StyledTableRow = styled(TableRow)`
+  td:first-child {
```
Comment (sohkai) [inline apps/finance/app/src/components/TransferRow.js]: Similar to https://github.com/aragon/aragon-apps/pull/722#discussion_r270040540.

File: apps/finance/app/src/components/TransferRow.js [hunk 6]
```diff
@@ -86,7 +90,11 @@ class TransferRow extends React.PureComponent {
           </time>
         </NoWrapCell>
         <NoWrapCell>
-          <IdentityBadge networkType={network.type} entity={entity} />
+          <LocalIdentityBadge
+            networkType={network.type}
+            entity={entity}
+            address={entity}
```
Comment (sohkai) [inline apps/finance/app/src/components/TransferRow.js]: Similar to https://github.com/aragon/aragon-apps/pull/722#discussion_r270042117.

File: apps/finance/app/package.json (hunk 4 above)
Comment (AquiGorka) [reply apps/finance/app/package.json]: Hmm, I'll remove it from this PR and maybe we can take care of it in a different one.

**Pull request merged**